static/app/components/narrowLayout.tsx

```diff
@@ -1,58 +1,54 @@
-import {Component} from 'react';
+import {useEffect} from 'react';
 import styled from '@emotion/styled';
 
 import {logout} from 'sentry/actionCreators/account';
-import {Client} from 'sentry/api';
 import {IconSentry} from 'sentry/icons';
 import {t} from 'sentry/locale';
+import useApi from 'sentry/utils/useApi';
 
-type Props = {
+type Props = React.PropsWithChildren<{
   maxWidth?: string;
   showLogout?: boolean;
-};
+}>;
 
-class NarrowLayout extends Component<Props> {
-  UNSAFE_componentWillMount() {
+function NarrowLayout({maxWidth, showLogout, children}: Props) {
+  const api = useApi();
+
+  useEffect(() => {
     document.body.classList.add('narrow');
-  }
 
-  componentWillUnmount() {
-    this.api.clear();
-    document.body.classList.remove('narrow');
+    return () => document.body.classList.remove('narrow');
+  }, []);
+
+  async function handleLogout() {
+    await logout(api);
+    window.location.assign('/auth/login');
```

Comment:
shouldn't we use the router here?

Reply:
nah auth login is not in react, it needs to actually reload the page.